azure-pipelines.yml

```diff
@@ -103,29 +127,53 @@ jobs:
 
   - script: |
       export LLVM_CONFIG_PATH=$(brew --prefix llvm)/bin/llvm-config
-      ./scripts/run_ci_checks.sh fmt
+      export RUSTFLAGS="-D warnings"
+      export RUSTDOCFLAGS="-D warnings"
+      cargo fmt --check
     displayName: 'cargo fmt --check'
   
   - script: |
       export LLVM_CONFIG_PATH=$(brew --prefix llvm)/bin/llvm-config
-      ./scripts/run_ci_checks.sh check
+      export RUSTFLAGS="-D warnings"
+      export RUSTDOCFLAGS="-D warnings"
+      # Soon to be superceded by the commented out `cargo clippy` below.
+      # This is different from `cargo build`
+      # as this uses `--all-features` to check everything.
+      cargo check --tests --all-features
+      # cargo clippy --tests --all-features
     displayName: 'cargo check'
 
   - script: |
       export LLVM_CONFIG_PATH=$(brew --prefix llvm)/bin/llvm-config
-      ./scripts/run_ci_checks.sh doc
+      export RUSTFLAGS="-D warnings"
+      export RUSTDOCFLAGS="-D warnings"
+      cargo doc --all-features --document-private-items --no-deps
     displayName: 'cargo doc'
 
   - script: |
       export LLVM_CONFIG_PATH=$(brew --prefix llvm)/bin/llvm-config
-      ./scripts/run_ci_checks.sh build
+      export RUSTFLAGS="-D warnings"
+      export RUSTDOCFLAGS="-D warnings"
+      # Don't build with `--all-features` as `--all-features` includes `--features llvm-static`,
```

Comment:
nitpick: You can cut down on duplication by saying something like 

```
# Don't build with `all-features`, see fast build step for Linux.
```

Reply:
Fixed in https://github.com/immunant/c2rust/pull/696/commits/da006571f57674fef171f13142881a4caf489c30.  I also made the same referring to the Linux comment for details change for the test translator step, too.